TESTS/mbed_hal/spm/main.cpp

```diff
@@ -84,7 +84,7 @@ __attribute__((naked)) void call_mem(uint32_t addr)
     // since exception will be generated for invalid memory access.
     // Other instructions are for calling do_nothing function according to AAPCS.
     __ASM(
-        "LDR     r1, [r0]\n"
+        "LDR     r4, [r0]\n"
```

Comment:
Since the compiler sees the definition of call_mem() when compiling test_memory(), it knows that r1 isn't used and doesn't need to be backed up. It would probably back it up if r1 was in the clobber list of the asm, as it should be.